pkg/gather/clusterconfig/operators_pods_and_events.go

```diff
@@ -250,33 +255,75 @@ func getContainerLogs(ctx context.Context,
 			continue
 		}
 
-		logName := fmt.Sprintf("%s_current.log", c.Name)
-		if isPrevious {
-			logName = fmt.Sprintf("%s_previous.log", c.Name)
-		}
+		logName := logFilename(c.Name, isPrevious)
 
 		buf.Reset()
 		klog.V(2).Infof("Fetching logs for %s container %s pod in namespace %s (previous: %v).", c.Name, pod.Name, pod.Namespace, isPrevious)
 		// Fetch container logs and continue on error
-		err := fetchPodContainerLog(ctx, client, pod, buf, c.Name, isPrevious, &bufferSize)
+		logString, err := getContainerLogString(ctx, client, pod, c.Name, isPrevious, buf, &bufferSize)
 		if err != nil {
 			klog.V(2).Infof("Error: %q", err)
 			continue
 		}
 
-		// Do not record empty records
-		if buf.Len() == 0 {
-			continue
+		// check for stack tracer
+		if found := stackTraceRegex.MatchString(logString); found {
+			klog.V(2).Infof("Stack trace found in log. Fetching the complete log for %s container in namespace %s (previous: %v).", c.Name, pod.Name, pod.Namespace, isPrevious)
+			fullLog, err := getContainerLogString(ctx, client, pod, c.Name, isPrevious, nil, nil)
+			if err != nil {
+				klog.V(2).Infof("Error: %q", err)
+				continue
+			}
+			logString = fullLog
 		}
-		records = append(records, record.Record{
-			Name: fmt.Sprintf("config/pod/%s/logs/%s/%s", pod.Namespace, pod.Name, logName),
-			Item: marshal.Raw{Str: buf.String()},
-		})
+
+		records = append(records, createLogRecord(pod.Namespace, pod.Name, logName, logString))
 	}
 
 	return records
 }
 
+// getContainerLogString fetch the container log from API and return it as String
+func getContainerLogString(
+	ctx context.Context,
+	client corev1client.CoreV1Interface,
+	pod *corev1.Pod,
+	name string,
+	isPrevious bool,
+	buf *bytes.Buffer,
+	bufferSize *int64) (string, error) {
+	if buf == nil {
+		buf = &bytes.Buffer{}
+	}
+	err := fetchPodContainerLog(ctx, client, pod, buf, name, isPrevious, bufferSize)
```

Comment:
Note: I know that this `fetchPodContainerLog` by default uses the `logTailLines` that will block the committed changes to work properly. I didn't change it yet, 'cause I'm exploring some ideas around it. :)